ledger/participant-integration-api/src/main/scala/platform/indexer/sequential/SequentialIndexer.scala

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.indexer.sequential
+
+import java.sql.Connection
+
+import com.daml.ledger.participant.state.v1.{Offset, Update}
+import com.daml.platform.store.appendonlydao.JdbcLedgerDao
+import com.daml.platform.store.backend.{DBDTOV1, StorageBackend}
+
+import scala.util.chaining.scalaUtilChainingOps
+
+trait SequentialIndexer {
+  def store(connection: Connection, offset: Offset, update: Option[Update]): Unit
+}
+
+object NoopSequentialIndexer extends SequentialIndexer {
+  override def store(connection: Connection, offset: Offset, update: Option[Update]): Unit =
+    throw new UnsupportedOperationException("NoopSequentialIndexer/store called. Not supported.")
+}
+
+case class SequentialIndexerImpl[DB_BATCH](
+    storageBackend: StorageBackend[DB_BATCH],
+    updateToDbDtos: Offset => Update => Iterator[DBDTOV1],
+) extends SequentialIndexer {
+
+  private var lastEventSeqId: Long = _
```

Comment:
yes it would be, but on the other hand it would obfuscate the code because everywhere we would need to do a .get or a pattern match from the lastEventSeqId. not sure which is better.
if we wouldn't have to do lazy init, it would be much nicer.